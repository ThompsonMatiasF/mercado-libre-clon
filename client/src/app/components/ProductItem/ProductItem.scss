@import "src/styles/custom";

li.item {
  padding: 16px 0;
  border-bottom: 1px solid $gray-200;

  & > img {
    width: 180px;
    min-width: 180px;
    height: 180px;
    margin-right: 16px;
    border-radius: 4px;
    box-sizing: content-box;
  }

  & > div {
    padding-top: 16px;
    p {
      font-size: 18px;
      line-height: 18px;
      margin-top: 32px;
    }

    .price {
      font-size: 24px;
      line-height: 24px;
    }

    .location {
      font-size: 12px;
      line-height: 12px;
      color: $gray-700;
    }
  }
}
