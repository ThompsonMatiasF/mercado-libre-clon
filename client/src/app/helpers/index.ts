export { default as mocks } from './mocks'
