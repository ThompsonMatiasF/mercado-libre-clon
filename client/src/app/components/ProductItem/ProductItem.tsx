import React, { FC } from 'react';
import { history } from 'app/router';
import './ProductItem.scss';
import { Item } from 'app/types';

interface ProductItemProps {
  data: Item
}

export const ProductItem: FC<ProductItemProps> = ({ data }) => (
  <li className="d-flex item" onClick={() => history.push(`/items/${data.id}`)}>
    <img src={data.picture} alt="product image" />
    <div className="flex-grow-1">
      <div className="d-flex justify-content-between align-items-center">
        <span className="d-flex align-items-center price">
          $
          {' '}
          {new Intl.NumberFormat('es-AR').format(data.price.amount)}
          {data.free_shipping
          && (
          <img
            src={`${process.env.PUBLIC_URL}/assets/images/shipping-min.png`}
            alt="free shipping icon"
            className="ml-2"
          />
          )}
        </span>
        <span className=" location">{ data.address }</span>
      </div>
      <p>{data.title}</p>
    </div>
  </li>
);

export default React.memo(ProductItem);
