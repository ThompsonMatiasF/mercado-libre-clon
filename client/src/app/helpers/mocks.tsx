import React, { FC } from 'react';
import { AppContext } from 'app/store';

interface MockContextProps {
  context: any
}

export const MockContext: FC<MockContextProps> = ({ context, children })  => (
  <AppContext.Provider value={ context }>{ children }</AppContext.Provider>
)

export const mockCategories = ['Celulares y Teléfonos', 'Celulares y Smartphones']

const mockPrice = {
  currency: 'ARS',
  amount: 100,
  decimals: 0,
}

export const mockItem = {
  id: 'MLA844506827',
  title: 'Iphone X 128gb',
  price: mockPrice,
  picture: 'https://domain.com/image.png',
  condition: 'new',
  free_shipping: true,
  address: 'Capital Federal',
}

export const mockItemDetail = {
  ...mockItem,
  categories: mockCategories,
  sold_quantity: 2,
  description: 'Iphone X 128g, nuevo, en caja cerrada con garantía oficial',
}

export const mockItems = Array(5).fill(mockItem)

export const mockSearchResult = {
  categories: mockCategories,
  items: mockItems,
}

export default {
  MockContext,
  mockItems,
  mockItem,
  mockItemDetail,
  mockSearchResult,
}

