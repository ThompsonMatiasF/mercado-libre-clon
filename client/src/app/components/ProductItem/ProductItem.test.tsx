import { render, screen } from '@testing-library/react'
import ProductItem from './ProductItem'
import { mockItem } from 'app/helpers/mocks';

describe('ProductItem.tsx', () => {
  it('Should match snapshot', () => {
    const { container } = render(<ProductItem data={mockItem} />)

    expect(container).toMatchSnapshot()
  })

  it.each([true, false])('Should evaluate free_shipping field to show shipping image', (freeShipping) => {
    const customMock = {...mockItem, free_shipping: freeShipping }
    const { container } = render(<ProductItem data={customMock} />)

    const expectation = expect(container.querySelectorAll('img').length)
    freeShipping ? expectation.toEqual(2) : expectation.toEqual(1)
  })
})
