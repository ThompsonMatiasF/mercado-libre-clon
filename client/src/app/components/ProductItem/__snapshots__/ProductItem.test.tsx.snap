// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem.tsx Should match snapshot 1`] = `
<div>
  <li
    class="d-flex item"
  >
    <img
      alt="product image"
      src="https://domain.com/image.png"
    />
    <div
      class="flex-grow-1"
    >
      <div
        class="d-flex justify-content-between align-items-center"
      >
        <span
          class="d-flex align-items-center price"
        >
          $
           
          100
          <img
            alt="free shipping icon"
            class="ml-2"
            src="/assets/images/shipping-min.png"
          />
        </span>
        <span
          class=" location"
        >
          Capital Federal
        </span>
      </div>
      <p>
        Iphone X 128gb
      </p>
    </div>
  </li>
</div>
`;
